<launch>
  <arg name="car" default="true"/>
  <arg name="pedestrian" default="false"/>
  <arg name="sync" default="false" />

  <group if="$(arg car)">
    <group ns="obj_car">

      <node pkg="cv_tracker" name="reprojection" type="obj_reproj">
        <remap from="/obj_car/image_obj_tracked" to="/sync_reprojection/obj_car/image_obj_tracked" if="$(arg sync)" />
        <remap from="/current_pose" to="/sync_reprojection/obj_car/current_pose" if="$(arg sync)" />
      </node>

    </group>
  </group>

  <group if="$(arg pedestrian)">
    <group ns="obj_person">

      <node pkg="cv_tracker" name="reprojection" type="obj_reproj">
        <remap from="/obj_person/image_obj_tracked" to="/sync_reprojection/obj_person/image_obj_tracked" if="$(arg sync)" />
        <remap from="/current_pose" to="/sync_reprojection/obj_person/current_pose" if="$(arg sync)" />
      </node>

    </group>
  </group>

  <include file="$(find synchronization)/launch/sync_obj_reproj.launch" if="$(arg sync)">
    <arg name="car" value="$(arg car)"/>
    <arg name="pedestrian" value="$(arg pedestrian)"/>
  </include>
</launch>
